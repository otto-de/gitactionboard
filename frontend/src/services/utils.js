import storageService from "@/services/storageService";

export const setAvailableAuths = (availableAuths) => {
  storageService.setItem("availableAuths", JSON.stringify(availableAuths));
};

export const setGithubSecretsScanMonitoringEnabled = (
  githubSecretsScanMonitoringEnabled
) => {
  storageService.setItem(
    "githubSecretsScanMonitoringEnabled",
    JSON.stringify(githubSecretsScanMonitoringEnabled)
  );
};

export const setGithubCodeScanMonitoringEnabled = (
  githubSecretsScanMonitoringEnabled
) => {
  storageService.setItem(
    "githubCodeScanMonitoringEnabled",
    JSON.stringify(githubSecretsScanMonitoringEnabled)
  );
};

export const getAvailableAuths = () =>
  JSON.parse(storageService.getItem("availableAuths"));

export const getGithubSecretsScanMonitoringEnabled = () =>
  JSON.parse(storageService.getItem("githubSecretsScanMonitoringEnabled")) ||
  false;

export const getGithubCodeScanMonitoringEnabled = () =>
  JSON.parse(storageService.getItem("githubCodeScanMonitoringEnabled")) ||
  false;
