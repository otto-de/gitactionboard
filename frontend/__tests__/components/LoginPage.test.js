import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { mount } from '../test-utils';
import LoginPage from '@/components/LoginPage.vue';

import { authenticate, fetchConfig } from '@/services/apiService';
import { isAuthenticate } from '@/services/authenticationService';
import {
  setAvailableAuths,
  setGithubCodeScanMonitoringEnabled,
  setGithubSecretsScanMonitoringEnabled,
  setVersion
} from '@/services/utils';

import flushPromises from 'flush-promises';
import Spinner from '@/components/Spinner.vue';
import getVuetify from '@/plugins/vuetify';
import preferences from '@/services/preferences';

vi.mock('@/services/apiService');
vi.mock('@/services/authenticationService');
vi.mock('@/services/utils');

describe('<LoginPage />', () => {
  const defaultConfig = {
    availableAuths: ['BASIC_AUTH'],
    githubSecretsScanMonitoringEnabled: true,
    githubCodeScanMonitoringEnabled: true,
    version: '3.3.0'
  };

  const render = (mockRouterPush = vi.fn()) =>
    mount(LoginPage, {
      global: {
        plugins: [getVuetify()],
        mocks: {
          $router: {
            push: mockRouterPush
          }
        }
      }
    });

  beforeEach(() => {
    vi.spyOn(preferences, 'theme', 'get').mockReturnValueOnce('light');
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should render spinner while fetching config', async () => {
    fetchConfig.mockResolvedValueOnce({
      availableAuths: ['BASIC_AUTH']
    });

    const vueWrapper = render();

    expect(vueWrapper.findComponent(Spinner).exists()).toBeTruthy();

    expect(vueWrapper.html()).toMatchSnapshot();

    await flushPromises();

    expect(vueWrapper.findComponent(Spinner).exists()).toBeFalsy();
    expect(fetchConfig).toHaveBeenCalledOnce();
  });

  it('should redirect to dashboard page if authentication is not configured', async () => {
    const config = {
      ...defaultConfig,
      availableAuths: [],
      githubSecretsScanMonitoringEnabled: false
    };
    fetchConfig.mockResolvedValueOnce(config);

    isAuthenticate.mockReturnValue(false);

    const mockRouterPush = vi.fn();
    const vueWrapper = render(mockRouterPush);

    await flushPromises();

    expect(vueWrapper.html()).toMatchSnapshot();

    expect(fetchConfig).toHaveBeenCalledOnce();
    expect(isAuthenticate).toHaveBeenCalledOnce();

    expect(setVersion).toHaveBeenCalledOnce();
    expect(setVersion).toHaveBeenCalledWith(config.version);

    expect(setAvailableAuths).toHaveBeenCalledOnce();
    expect(setAvailableAuths).toHaveBeenCalledWith(config.availableAuths);

    expect(setGithubSecretsScanMonitoringEnabled).toHaveBeenCalledOnce();
    expect(setGithubSecretsScanMonitoringEnabled).toHaveBeenCalledWith(config.githubSecretsScanMonitoringEnabled);
    expect(setGithubCodeScanMonitoringEnabled).toHaveBeenCalledOnce();
    expect(setGithubCodeScanMonitoringEnabled).toHaveBeenCalledWith(config.githubCodeScanMonitoringEnabled);
    expect(mockRouterPush).toHaveBeenCalledOnce();
    expect(mockRouterPush).toHaveBeenCalledWith('workflow-jobs');
  });

  it.each([
    [['BASIC_AUTH']],
    [['OAUTH2']],
    [['BASIC_AUTH', 'OAUTH2']]
  ])('should render login page only with auth mechanism %s', async (availableAuths) => {
    const config = {
      ...defaultConfig,
      availableAuths,
      githubCodeScanMonitoringEnabled: false
    };
    fetchConfig.mockResolvedValueOnce(config);

    isAuthenticate.mockReturnValue(false);

    const vueWrapper = render();

    await flushPromises();

    expect(vueWrapper.html()).toMatchSnapshot();

    expect(fetchConfig).toHaveBeenCalledOnce();
    expect(isAuthenticate).toHaveBeenCalledOnce();

    expect(setVersion).toHaveBeenCalledOnce();
    expect(setVersion).toHaveBeenCalledWith(config.version);

    expect(setAvailableAuths).toHaveBeenCalledOnce();
    expect(setAvailableAuths).toHaveBeenCalledWith(config.availableAuths);

    expect(setGithubSecretsScanMonitoringEnabled).toHaveBeenCalledOnce();
    expect(setGithubSecretsScanMonitoringEnabled).toHaveBeenCalledWith(config.githubSecretsScanMonitoringEnabled);
    expect(setGithubCodeScanMonitoringEnabled).toHaveBeenCalledOnce();
    expect(setGithubCodeScanMonitoringEnabled).toHaveBeenCalledWith(config.githubCodeScanMonitoringEnabled);
  });

  describe('Basic Auth', () => {
    let mockRouterPush;
    let vueWrapper;

    const username = 'admin';
    const password = 'password';

    const basicAuthLoginForm = '[test-id="basic-auth-login-form"]';
    const basicAuthLoginButton = '[test-id="basic-auth-login-button"]';
    const usernameInputField = '[test-id="username"]';
    const passwordInputField = '[test-id="password"]';
    const basicAuthAlert = '[test-id="basic-auth-alert"]';
    const vBtnDisabledClassName = 'v-btn--disabled';
    const submitPreventEventName = 'submit.prevent';

    beforeEach(async () => {
      const config = {
        ...defaultConfig,
        githubSecretsScanMonitoringEnabled: false,
        githubCodeScanMonitoringEnabled: false
      };
      fetchConfig.mockResolvedValueOnce(config);

      isAuthenticate.mockReturnValue(false);

      mockRouterPush = vi.fn();

      vueWrapper = render(mockRouterPush);

      await flushPromises();

      expect(vueWrapper.find(basicAuthLoginButton).classes(vBtnDisabledClassName)).toBeTruthy();
      await vueWrapper.find(usernameInputField).setValue(username);
      expect(vueWrapper.find(basicAuthLoginButton).classes(vBtnDisabledClassName)).toBeTruthy();
      await vueWrapper.find(passwordInputField).setValue(password);

      expect(vueWrapper.find(basicAuthLoginButton).classes(vBtnDisabledClassName)).toBeFalsy();
    });

    it('should redirect to dashboard post successful login', async () => {
      authenticate.mockResolvedValueOnce({});

      await vueWrapper.find(basicAuthLoginForm).trigger(submitPreventEventName);

      await flushPromises();

      expect(authenticate).toHaveBeenCalledOnce();
      expect(authenticate).toHaveBeenCalledWith(username, password);

      expect(mockRouterPush).toHaveBeenCalledOnce();
      expect(mockRouterPush).toHaveBeenCalledWith('workflow-jobs');
    });

    it('should not redirect to dashboard for unsuccessful login', async () => {
      authenticate.mockRejectedValueOnce('error');

      await vueWrapper.find(basicAuthLoginForm).trigger(submitPreventEventName);

      await flushPromises();

      expect(vueWrapper.find(basicAuthAlert).exists()).toBeTruthy();

      expect(authenticate).toHaveBeenCalledOnce();
      expect(authenticate).toHaveBeenCalledWith(username, password);

      expect(mockRouterPush).not.toBeCalled();
    });

    it('should hide alert once user provide new username post unsuccessful login', async () => {
      authenticate.mockRejectedValueOnce('error');

      await vueWrapper.find(basicAuthLoginForm).trigger(submitPreventEventName);

      await flushPromises();

      expect(vueWrapper.find(basicAuthAlert).exists()).toBeTruthy();

      await vueWrapper.find(usernameInputField).setValue(username + '_new');
      expect(vueWrapper.find(basicAuthAlert).exists()).toBeFalsy();
      expect(vueWrapper.find(basicAuthLoginButton).classes(vBtnDisabledClassName)).toBeFalsy();
    });

    it('should hide alert once user provide new password post unsuccessful login', async () => {
      authenticate.mockRejectedValueOnce('error');

      await vueWrapper.find(basicAuthLoginForm).trigger(submitPreventEventName);

      await flushPromises();

      expect(vueWrapper.find(basicAuthAlert).exists()).toBeTruthy();

      await vueWrapper.find(passwordInputField).setValue(password + '_new');
      expect(vueWrapper.find(basicAuthAlert).exists()).toBeFalsy();
      expect(vueWrapper.find(basicAuthLoginButton).classes(vBtnDisabledClassName)).toBeFalsy();
    });
  });
});
