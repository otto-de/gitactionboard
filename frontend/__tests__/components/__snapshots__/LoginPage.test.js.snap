// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LoginPage /> > should redirect to dashboard page if authentication is not configured 1`] = `
<div class="v-container v-container--fluid v-locale--is-ltr h-screen fill-height justify-center main-container">
  <!--v-if-->
  <div class="v-card v-theme--light v-card--density-default elevation-5 v-card--rounded v-card--variant-elevated px-5 py-10 rounded-xl" style="min-width: 33%;">
    <!---->
    <div class="v-card__loader">
      <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
        <!---->
        <div class="v-progress-linear__background"></div>
        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
        <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class="v-card-title header text-center"> Gitaction Board </div>
    <div class="v-card-subtitle welcome-message text-center mb-10 mt-1"> Welcome to Gitaction Board </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!---->
    <!---->
    <span class="v-card__underlay"></span>
  </div>
</div>
`;

exports[`<LoginPage /> > should render login page only with auth mechanism BASIC_AUTH 1`] = `
<div class="v-container v-container--fluid v-locale--is-ltr h-screen fill-height justify-center main-container">
  <!--v-if-->
  <div class="v-card v-theme--light v-card--density-default elevation-5 v-card--rounded v-card--variant-elevated px-5 py-10 rounded-xl" style="min-width: 33%;">
    <!---->
    <div class="v-card__loader">
      <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
        <!---->
        <div class="v-progress-linear__background"></div>
        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
        <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class="v-card-title header text-center"> Gitaction Board </div>
    <div class="v-card-subtitle welcome-message text-center mb-10 mt-1"> Welcome to Gitaction Board </div>
    <div class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <form class="v-form" novalidate test-id="basic-auth-login-form">
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field mb-2">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator>
                  <label aria-hidden="true" class="v-label v-field-label v-field-label--floating" for="input-0">
                    <!---->
                    Username
                  </label>
                  <label class="v-label v-field-label" for="input-0">
                    <!---->
                    Username
                  </label>
                  <!---->
                  <input aria-describedby="input-0-messages" class="v-field__input" id="input-0" placeholder="Enter your username" size="1" test-id="username" type="text" value>
                  <!---->
                </div>
                <transition-stub appear="false" css="true" name="expand-x-transition" persisted="false">
                  <div class="v-field__clearable" style="display: none;">
                    <i aria-hidden="false" aria-label="Clear Username" class="v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" tabindex="0">
                      <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                        <path d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z" />
                      </svg>
                    </i>
                  </div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!---->
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator>
                  <label aria-hidden="true" class="v-label v-field-label v-field-label--floating" for="input-2">
                    <!---->
                    Password
                  </label>
                  <label class="v-label v-field-label" for="input-2">
                    <!---->
                    Password
                  </label>
                  <!---->
                  <input aria-describedby="input-2-messages" class="v-field__input" id="input-2" placeholder="Enter your password" size="1" test-id="password" type="password" value>
                  <!---->
                </div>
                <transition-stub appear="false" css="true" name="expand-x-transition" persisted="false">
                  <div class="v-field__clearable" style="display: none;">
                    <i aria-hidden="false" aria-label="Clear Password" class="v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" tabindex="0">
                      <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                        <path d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z" />
                      </svg>
                    </i>
                  </div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!---->
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
          <!---->
          <button class="v-btn v-btn--block v-btn--disabled v-theme--light bg-success v-btn--density-default v-btn--size-large v-btn--variant-elevated mt-4" disabled test-id="basic-auth-login-button" type="submit">
            <span class="v-btn__overlay"></span>
            <span class="v-btn__underlay"></span>
            <!---->
            <span class="v-btn__content" data-no-activator> Login </span>
            <span class="v-btn__append">
              <i aria-hidden="true" class="v-icon notranslate v-theme--light v-icon--size-default">
                <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                  <path d="M11 7L9.6 8.4L12.2 11H2V13H12.2L9.6 15.6L11 17L16 12L11 7M20 19H12V21H20C21.1 21 22 20.1 22 19V5C22 3.9 21.1 3 20 3H12V5H20V19Z" />
                </svg>
              </i>
            </span>
            <!---->
          </button>
        </form>
      </div>
      <!---->
    </div>
    <!--v-if-->
    <!--v-if-->
    <!---->
    <!---->
    <span class="v-card__underlay"></span>
  </div>
</div>
`;

exports[`<LoginPage /> > should render login page only with auth mechanism BASIC_AUTH,OAUTH2 1`] = `
<div class="v-container v-container--fluid v-locale--is-ltr h-screen fill-height justify-center main-container">
  <!--v-if-->
  <div class="v-card v-theme--light v-card--density-default elevation-5 v-card--rounded v-card--variant-elevated px-5 py-10 rounded-xl" style="min-width: 33%;">
    <!---->
    <div class="v-card__loader">
      <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
        <!---->
        <div class="v-progress-linear__background"></div>
        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
        <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class="v-card-title header text-center"> Gitaction Board </div>
    <div class="v-card-subtitle welcome-message text-center mb-10 mt-1"> Welcome to Gitaction Board </div>
    <div class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <form class="v-form" novalidate test-id="basic-auth-login-form">
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field mb-2">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator>
                  <label aria-hidden="true" class="v-label v-field-label v-field-label--floating" for="input-0">
                    <!---->
                    Username
                  </label>
                  <label class="v-label v-field-label" for="input-0">
                    <!---->
                    Username
                  </label>
                  <!---->
                  <input aria-describedby="input-0-messages" class="v-field__input" id="input-0" placeholder="Enter your username" size="1" test-id="username" type="text" value>
                  <!---->
                </div>
                <transition-stub appear="false" css="true" name="expand-x-transition" persisted="false">
                  <div class="v-field__clearable" style="display: none;">
                    <i aria-hidden="false" aria-label="Clear Username" class="v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" tabindex="0">
                      <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                        <path d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z" />
                      </svg>
                    </i>
                  </div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!---->
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator>
                  <label aria-hidden="true" class="v-label v-field-label v-field-label--floating" for="input-2">
                    <!---->
                    Password
                  </label>
                  <label class="v-label v-field-label" for="input-2">
                    <!---->
                    Password
                  </label>
                  <!---->
                  <input aria-describedby="input-2-messages" class="v-field__input" id="input-2" placeholder="Enter your password" size="1" test-id="password" type="password" value>
                  <!---->
                </div>
                <transition-stub appear="false" css="true" name="expand-x-transition" persisted="false">
                  <div class="v-field__clearable" style="display: none;">
                    <i aria-hidden="false" aria-label="Clear Password" class="v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" tabindex="0">
                      <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                        <path d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z" />
                      </svg>
                    </i>
                  </div>
                </transition-stub>
                <div class="v-field__append-inner">
                  <!---->
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
          <!---->
          <button class="v-btn v-btn--block v-btn--disabled v-theme--light bg-success v-btn--density-default v-btn--size-large v-btn--variant-elevated mt-4" disabled test-id="basic-auth-login-button" type="submit">
            <span class="v-btn__overlay"></span>
            <span class="v-btn__underlay"></span>
            <!---->
            <span class="v-btn__content" data-no-activator> Login </span>
            <span class="v-btn__append">
              <i aria-hidden="true" class="v-icon notranslate v-theme--light v-icon--size-default">
                <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                  <path d="M11 7L9.6 8.4L12.2 11H2V13H12.2L9.6 15.6L11 17L16 12L11 7M20 19H12V21H20C21.1 21 22 20.1 22 19V5C22 3.9 21.1 3 20 3H12V5H20V19Z" />
                </svg>
              </i>
            </span>
            <!---->
          </button>
        </form>
      </div>
      <!---->
    </div>
    <div class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <div class="v-row align-center my-3">
          <div class="v-col v-col-5">
            <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 3px;">
          </div>
          <div class="v-col text-center font-weight-bold"> OR </div>
          <div class="v-col v-col-5">
            <hr aria-orientation="horizontal" class="v-divider v-theme--light" role="separator" style="border-top-width: 3px;">
          </div>
        </div>
      </div>
      <!---->
    </div>
    <div class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <button class="v-btn v-btn--block v-btn--elevated v-theme--light bg-success v-btn--density-default v-btn--size-large v-btn--variant-elevated" type="submit">
          <span class="v-btn__overlay"></span>
          <span class="v-btn__underlay"></span>
          <!---->
          <span class="v-btn__content" data-no-activator> Login with Github </span>
          <span class="v-btn__append">
            <i aria-hidden="true" class="v-icon notranslate v-theme--light v-icon--size-default">
              <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                <path d="M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z" />
              </svg>
            </i>
          </span>
          <!---->
        </button>
      </div>
      <!---->
    </div>
    <!---->
    <!---->
    <span class="v-card__underlay"></span>
  </div>
</div>
`;

exports[`<LoginPage /> > should render login page only with auth mechanism OAUTH2 1`] = `
<div class="v-container v-container--fluid v-locale--is-ltr h-screen fill-height justify-center main-container">
  <!--v-if-->
  <div class="v-card v-theme--light v-card--density-default elevation-5 v-card--rounded v-card--variant-elevated px-5 py-10 rounded-xl" style="min-width: 33%;">
    <!---->
    <div class="v-card__loader">
      <div aria-hidden="true" aria-valuemax="100" aria-valuemin="0" class="v-progress-linear v-theme--light v-locale--is-ltr" role="progressbar" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;">
        <!---->
        <div class="v-progress-linear__background"></div>
        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
        <transition-stub appear="false" css="true" name="fade-transition" persisted="false">
          <div class="v-progress-linear__indeterminate">
            <div class="v-progress-linear__indeterminate long"></div>
            <div class="v-progress-linear__indeterminate short"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class="v-card-title header text-center"> Gitaction Board </div>
    <div class="v-card-subtitle welcome-message text-center mb-10 mt-1"> Welcome to Gitaction Board </div>
    <!--v-if-->
    <!--v-if-->
    <div class="v-card-item">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <button class="v-btn v-btn--block v-btn--elevated v-theme--light bg-success v-btn--density-default v-btn--size-large v-btn--variant-elevated" type="submit">
          <span class="v-btn__overlay"></span>
          <span class="v-btn__underlay"></span>
          <!---->
          <span class="v-btn__content" data-no-activator> Login with Github </span>
          <span class="v-btn__append">
            <i aria-hidden="true" class="v-icon notranslate v-theme--light v-icon--size-default">
              <svg aria-hidden="true" class="v-icon__svg" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                <path d="M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z" />
              </svg>
            </i>
          </span>
          <!---->
        </button>
      </div>
      <!---->
    </div>
    <!---->
    <!---->
    <span class="v-card__underlay"></span>
  </div>
</div>
`;

exports[`<LoginPage /> > should render spinner while fetching config 1`] = `
<div class="v-container v-container--fluid v-locale--is-ltr h-screen fill-height justify-center main-container">
  <div aria-valuemax="100" aria-valuemin="0" class="v-progress-circular v-progress-circular--indeterminate v-theme--light" role="progressbar" style="width: 128px; height: 128px; color: rgb(95, 141, 119); caret-color: #5f8d77;">
    <svg style="transform: rotate(calc(-90deg + 0deg));" viewBox="0 0 44.13793103448276 44.13793103448276" xmlns="http://www.w3.org/2000/svg">
      <circle class="v-progress-circular__underlay" cx="50%" cy="50%" fill="transparent" r="20" stroke-dasharray="125.66370614359172" stroke-dashoffset="0" stroke-width="4.137931034482759" />
      <circle class="v-progress-circular__overlay" cx="50%" cy="50%" fill="transparent" r="20" stroke-dasharray="125.66370614359172" stroke-dashoffset="125.66370614359172px" stroke-width="4.137931034482759" />
    </svg>
    <!---->
  </div>
  <!--v-if-->
</div>
`;
